src/geo/layer_definition.js

```diff
@@ -911,7 +911,7 @@ NamedMap.prototype = _.extend({}, MapBase.prototype, {
     this.layers = _.clone(named_map.layers) || [];
     for(var i = 0; i < this.layers.length; ++i) {
       var layer = this.layers[i];
-      layer.options = layer.options || { hidden: false };
+      layer.options = layer.options || { 'hidden': layer.visible === false };
```

Comment:
Layers inside of the `named_map` key has a `visible` attribute which should determine the visibility of layers. It was broken until now.